### ISSUE TITLE
Default interface check

### ISSUE THREAD

**Issue by BertrandGouny** [acassen/keepalived#963]
Hello,
i'm having trouble with the default interface config : 
```
Thu Jul 26 09:35:47 2018: Starting Keepalived v2.0.6 (07/23,2018)
Thu Jul 26 09:35:47 2018: Running on Linux 4.14.55-coreos #1 SMP Tue Jul 24 23:42:33 UTC 2018 (built for Linux 4.4.6)
Thu Jul 26 09:35:47 2018: Opening file '/usr/local/etc/keepalived/keepalived.conf'.
Thu Jul 26 09:35:47 2018: WARNING - default interface (null) doesn't exist
Thu Jul 26 09:35:47 2018: Starting VRRP child process, pid=669
Thu Jul 26 09:35:47 2018: Registering Kernel netlink reflector
Thu Jul 26 09:35:47 2018: Registering Kernel netlink command channel
Thu Jul 26 09:35:47 2018: Opening file '/usr/local/etc/keepalived/keepalived.conf'.
Thu Jul 26 09:35:47 2018: Unknown keyword 'garp_lower_priority_repeat'
Thu Jul 26 09:35:47 2018: WARNING - default user 'keepalived_script' for script execution does not exist - please create.
Thu Jul 26 09:35:47 2018: Non-existent interface specified in configuration
Thu Jul 26 09:35:47 2018: Stopped
Thu Jul 26 09:35:47 2018: Keepalived_vrrp exited with permanent error CONFIG. Terminating
Thu Jul 26 09:35:47 2018: Stopping
Thu Jul 26 09:35:47 2018: Stopped Keepalived v2.0.6 (07/23,2018)
```

```
global_defs {
  default_interface tun0
}

vrrp_instance VI_1 {
  interface tun0

  track_interface {
    tun0
  }

  state BACKUP
  virtual_router_id 51

  priority 110
  nopreempt

  garp_master_repeat 0
  garp_lower_priority_repeat 0
  garp_master_refresh 0
  garp_master_refresh_repeat 0

  unicast_peer {
                  10.244.43.1
                  10.244.46.1
  }

  virtual_ipaddress {
    XX.XX.XX.XX/32 dev eno1
  }

  authentication {
    auth_type PASS
    auth_pass xxxxxx
  }
}
```

 ```
tun0      Link encap:UNSPEC  HWaddr 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  
          inet addr:10.244.73.0  P-t-P:10.244.73.0  Mask:255.255.0.0
          inet6 addr: fe80::793f:775a:19ac:e638/64 Scope:Link
          UP POINTOPOINT RUNNING NOARP MULTICAST  MTU:1500  Metric:1
          RX packets:219279 errors:0 dropped:0 overruns:0 frame:0
          TX packets:198814 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:500 
          RX bytes:168489066 (160.6 MiB)  TX bytes:86930237 (82.9 MiB)
```

not sure what is wrong with the interface :s

**Comment by pqarmitage** [comment]
The handling of default_interface was wrong in the code. Commit 2bb86a1 fixes it.

It appears that `default_interface` is only used for static_ipaddress configuration, and is not used for vrrp_instances, so it is of limited use (I don't know why I implemented it like that!).

You don't need to specify
```
  track_interface {
    tun0
  }
```
since the interface of a vrrp instance is tracked by default.

BTW `garp_lower_priority_repeat` should be `garp_lower_prio_repeat`. It was wrong in keepalived.conf.SYNOPSIS and is corrected in commit 8476495.

**Comment by BertrandGouny** [comment]
thanks :)
